### PR TITLE
Bumps pulpcore to 3.4, bumps plugin client versions.

### PR DESCRIPTION
**Pull request by jjeffers**

### CODE REVIEW

**Comment by jlsherrill** [comment]
missing a bump of release and a changelog entry :)

**Comment by snecklifter** [comment]
```DEBUG util.py:598:  Error: No Package found for tfm-rubygem(pulp_2to3_migration_client) >= 0.2.0b2
DEBUG util.py:598:  Error: No Package found for tfm-rubygem(pulp_container_client) >= 1.4.0
DEBUG util.py:598:  Error: No Package found for tfm-rubygem(pulp_file_client) >= 1.0.0
DEBUG util.py:598:  Error: No Package found for tfm-rubygem(pulp_rpm_client) >= 3.4.0
DEBUG util.py:598:  Error: No Package found for tfm-rubygem(pulpcore_client) >= 3.4.0```

**Comment by evgeni** [comment]
[test rpm]

File: packages/katello/rubygem-katello/rubygem-katello.spec [hunk 1]
```diff
@@ -75,18 +75,18 @@ BuildRequires: %{?scl_prefix}rubygem(apipie-rails) >= 0.5.14
 BuildRequires: %{?scl_prefix}rubygem(runcible) >= 2.13.0
 BuildRequires: %{?scl_prefix}rubygem(runcible) < 3.0.0
 BuildRequires: %{?scl_prefix}rubygem(anemone)
-BuildRequires: %{?scl_prefix}rubygem(pulpcore_client) >= 3.3.0
+BuildRequires: %{?scl_prefix}rubygem(pulpcore_client) >= 3.4.0
 BuildRequires: %{?scl_prefix}rubygem(pulpcore_client) < 3.4.0
```
Comment (evgeni) [inline packages/katello/rubygem-katello/rubygem-katello.spec:79]:
```suggestion
BuildRequires: %{?scl_prefix}rubygem(pulpcore_client) < 3.5.0
```

File: packages/katello/rubygem-katello/rubygem-katello.spec [hunk 2]
```diff
@@ -42,18 +42,18 @@ Requires: %{?scl_prefix}rubygem(apipie-rails) >= 0.5.14
...
-Requires: %{?scl_prefix}rubygem(pulp_container_client) >= 1.3.0
-Requires: %{?scl_prefix}rubygem(pulp_container_client) < 1.4.0
-Requires: %{?scl_prefix}rubygem(pulp_rpm_client) >= 3.3.0
-Requires: %{?scl_prefix}rubygem(pulp_rpm_client) < 3.4.0
-Requires: %{?scl_prefix}rubygem(pulp_2to3_migration_client) >= 0.2.0
+Requires: %{?scl_prefix}rubygem(pulp_container_client) >= 1.4.0
+Requires: %{?scl_prefix}rubygem(pulp_container_client) < 1.5.0
+Requires: %{?scl_prefix}rubygem(pulp_rpm_client) >= 3.4.0
+Requires: %{?scl_prefix}rubygem(pulp_rpm_client) < 3.5.0
+Requires: %{?scl_prefix}rubygem(pulp_2to3_migration_client) >= 0.2.0b2
```
Comment (evgeni) [inline packages/katello/rubygem-katello/rubygem-katello.spec:56]:
```suggestion
Requires: %{?scl_prefix}rubygem(pulp_2to3_migration_client) >= 0.2.0
```

(our RPM packages don't/can't follow GEM versioning)

File: packages/katello/rubygem-katello/rubygem-katello.spec [hunk 3]
```diff
@@ -75,18 +75,18 @@ BuildRequires: %{?scl_prefix}rubygem(apipie-rails) >= 0.5.14
...
-BuildRequires: %{?scl_prefix}rubygem(pulp_container_client) >= 1.3.0
-BuildRequires: %{?scl_prefix}rubygem(pulp_container_client) < 1.4.0
-BuildRequires: %{?scl_prefix}rubygem(pulp_rpm_client) >= 3.3.0
-BuildRequires: %{?scl_prefix}rubygem(pulp_rpm_client) < 3.4.0
-BuildRequires: %{?scl_prefix}rubygem(pulp_2to3_migration_client) >= 0.2.0
+BuildRequires: %{?scl_prefix}rubygem(pulp_container_client) >= 1.4.0
+BuildRequires: %{?scl_prefix}rubygem(pulp_container_client) < 1.5.0
+BuildRequires: %{?scl_prefix}rubygem(pulp_rpm_client) >= 3.4.0
+BuildRequires: %{?scl_prefix}rubygem(pulp_rpm_client) < 3.5.0
+BuildRequires: %{?scl_prefix}rubygem(pulp_2to3_migration_client) >= 0.2.0b2
```
Comment (evgeni) [inline packages/katello/rubygem-katello/rubygem-katello.spec:89]:
```suggestion
BuildRequires: %{?scl_prefix}rubygem(pulp_2to3_migration_client) >= 0.2.0
```

(see above)

**Comment by evgeni** [comment]
closing in favor of #5368

**Pull request closed without merging**